### ISSUE TITLE
How to define a function in falcor's JSON Graph?

### ISSUE THREAD

**Issue by leozdgao** [Netflix/falcor#769]
I'm very interested in this Modal Layer Library --- Falcor. And I read the official document of Falcor, and in [JSON Graph section](http://netflix.github.io/falcor/documentation/jsongraph.html), a topic of **The Abstract JSON Graph Operations**, and when it is talking about `call` operation, I'm totally confused. Here is my code according the official example:

```
var jsonGraphObject = {
    todosById: {
        "44": {
            name: "get milk from corner store",
            addedAt: 29689724399,
            done: false,
            prerequisites: [{ $type: "ref", value: ["todosById", 54] }]
        },
        "54": {
            name: "withdraw money from ATM",
            addedAt: 15687384689,
            done: false,
            prerequisites: []
        }
    },
    todos: [
        { $type: "ref", value: ["todosById", 44] },
        { $type: "ref", value: ["todosById", 54] }
    ]
};
var model = new falcor.Model({
  cache: jsonGraphObject
});
```

Then the document showed me about how to use `call`:

```
model.call(
    // callPath
    ["todos", "add"],
    // arguments
    ["pick up car from the shop"], 
    // refPaths
    [
        ["addedAt"]
    ],
    // thisPaths
    [
        ["length"]
    ])
    .then(console.log.bind(console))
```

And the document just show me the result, but my code did not work, I totally do not know where is this `add` function, and I don't know how to define a function in Falcor's JSON Graph.

I googled about it, but haven't get my answer.So is there any example or boilerplate project there? Thanks.

**Comment by greim** [comment]
> So is there any example or boilerplate project there? 

I think there is and it was definitely helpful for me, but I can't find the link to it now. I seem to recall it was mentioned in another issue asking how to use `call()`. Maybe search issues in either this repo or falcor-router.

**Comment by sdesai** [comment]
This may help get you started, if you're looking for a concrete example of how you'd set up a call handler on your router: https://github.com/Netflix/falcor-router-demo/blob/master/index.js#L457

IIRC, I don't believe local `calls` are supported, but I may be wrong [ that is, you won't be able to define a call-able function in your local cache object (the jsonGraphObject you set up above) ]. You'll need to involve a router/datasource.

**Comment by joshdmiller** [comment]
Check out the express demo [here](https://github.com/Netflix/falcor-express-demo), along with the router it uses [here](https://github.com/Netflix/falcor-router-demo). There are calls there.

Also there is a [section in the docs](http://netflix.github.io/falcor/documentation/model.html#calling-functions) that covers both sides.

**Comment by steveorsomethin** [comment]
I'm currently performing issue triage as we get ready to perform a proper release, and closing/tagging as I go.

Looks like this got good answers. Closing.
